tests/hardhat/Comptroller/setters.ts

```diff
@@ -115,4 +133,62 @@ describe("setters", async () => {
         .withArgs(owner.address, comptroller.address, "setMinLiquidatableCollateral(uint256)");
     });
   });
+
+  describe("rewardDistributor", async () => {
+    const newMinLiquidatableCollateral = convertToUnit("100", 18);
+    it("reverts if access control manager does not allow the call", async () => {
+      accessControl.isAllowedToCall
+        .whenCalledWith(owner.address, "setMinLiquidatableCollateral(uint256)")
+        .returns(false);
+      await expect(comptroller.setMinLiquidatableCollateral(newMinLiquidatableCollateral))
+        .to.be.revertedWithCustomError(comptroller, "Unauthorized")
+        .withArgs(owner.address, comptroller.address, "setMinLiquidatableCollateral(uint256)");
+    });
+  });
+
+  describe("SupplyAndBorrowCaps", async () => {
+    it("reverts if token data is invalid", async () => {
+      await expect(comptroller.setMarketSupplyCaps([], [1, 2])).to.be.revertedWith("invalid number of markets");
+    });
+
+    it("reverts if supply and token data is invalid", async () => {
+      await expect(comptroller.setMarketSupplyCaps([OMG.address], [1, 2])).to.be.revertedWith(
+        "invalid number of markets",
+      );
+    });
+
+    it("reverts if borrow and token data is invalid", async () => {
+      await expect(comptroller.setMarketBorrowCaps([OMG.address], [1, 2])).to.be.revertedWith("invalid input");
+    });
+  });
+
+  describe("setCollateralFactor", async () => {
+    it("reverts if market is not listed", async () => {
+      await expect(comptroller.setCollateralFactor(OMG.address, convertToUnit("0.8", 18), convertToUnit("0.7", 18)))
+        .to.be.revertedWithCustomError(comptroller, "MarketNotListed")
+        .withArgs(OMG.address);
+    });
+
+    it("reverts if collateral factor is greater then max collateral factor", async () => {
+      await comptroller.connect(poolRegistrySigner).supportMarket(OMG.address);
+      await expect(
+        comptroller.setCollateralFactor(OMG.address, convertToUnit("1", 18), convertToUnit("0.7", 18)),
+      ).to.be.revertedWithCustomError(comptroller, "InvalidCollateralFactor");
+    });
+
+    it("reverts if liquidation threshold is greater then collateral factor", async () => {
+      await comptroller.connect(poolRegistrySigner).supportMarket(OMG.address);
+      await expect(
+        comptroller.setCollateralFactor(OMG.address, convertToUnit("0.7", 18), convertToUnit("0.8", 18)),
+      ).to.be.revertedWithCustomError(comptroller, "InvalidLiquidationThreshold");
+    });
+
+    it("reverts if token price is zero", async () => {
+      await oracle.getUnderlyingPrice.returns(0);
```

Comment:
```suggestion
      oracle.getUnderlyingPrice.returns(0);
```
no need to await the faking of return value of a fake contract

Reply:
Fixed